### DISCUSSION TITLE
Updating Old Rows and Adding New Rows at the same time

### DISCUSSION THREAD

**Discussion by quantdev05** [supabase/supabase#14832] [Questions]
Hi there! 

I just started using Supabase (w/ supabase py) this week, but was wondering whether it is possible to implement the following description. 

So basically, I am using a python based web scraping API that "streams" new data which is supposed to add new rows , and every 5 minutes of its creation time, few columns of that row is updated with new data. Can I make this possible using supabase-py?

Many thanks for your support in advance, I really hope I can figure this problem. 

Best! Nick

**Comment by Charlotte-br560** [comment]
Yes, you can achieve that with Supabase. You can use the upsert function in Supabase to either insert a new row or update an existing one based on certain conditions. You'll need to structure your data and code accordingly. Additionally, I'd recommend checking out [Crawlbase](https://crawlbase.com/) for web scraping tasks as it integrates well with Supabase. Best of luck, Nick!
